### PR TITLE
Make mac-test "required" on PRs.

### PR DESCRIPTION
**Pull request by gen740**
## Motivation

Make mac-test "required" on PRs.

## Description of the changes

- Add scheduled test to the mac-test.

### CODE REVIEW

**Comment by gen740** [comment]
Make sure integration tests don't run on PRs.

File: .github/workflows/mac-tests.yml
Comment (eukaryo) [inline .github/workflows/mac-tests.yml:64]:
```suggestion
      if:  ${{ github.event_name != 'schedule' && github.event_name != 'workflow_dispatch' }}
```

**Review by eukaryo — COMMENTED** [review]
Almost LGTM; there seems to be a case which pytest will be executed in duplicate.

File: .github/workflows/mac-tests.yml
Comment (eukaryo) [inline .github/workflows/mac-tests.yml:13]: Good catch! Thank you.

**Review by eukaryo — APPROVED** [review]
LGTM!

File: .github/workflows/mac-tests.yml
Comment (HideakiImamura) [inline .github/workflows/mac-tests.yml:74]: This type of condition in [`tests-integration.yml`](https://github.com/optuna/optuna/blob/master/.github/workflows/tests-integration.yml#L18) is `(A && B) || B`. Should we use the same condition?

Comment (HideakiImamura) [reply .github/workflows/mac-tests.yml:74]: And, this should be applied to the `tests-mac` job.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have some comments on the conditions that limit the execution of tests on forked repos.

File: .github/workflows/mac-tests.yml
Comment (toshihikoyanase) [inline .github/workflows/mac-tests.yml:24]: I think the above comment is not consistent with this condition since non-scheduled workflows are expected to be executed in fork repositories. We can see a similar change in L74.

Comment (gen740) [reply .github/workflows/mac-tests.yml]: I update the comment.

Comment (gen740) [reply .github/workflows/mac-tests.yml]: The same conditions appear on Tests.yml (and other tests). Is it better to add the comment to them?

Comment (toshihikoyanase) [reply .github/workflows/mac-tests.yml]: Thank you for your update. I think we should work on `Tests.yml` in a follow-up PR to minimize the changes in this PR.

**Review by HideakiImamura — APPROVED** [review]
Looks very good.

**Comment by HideakiImamura** [comment]
@toshihikoyanase Please merge this PR after your approval.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for letting me know. LGTM!

**Pull request merged**
